### ISSUE TITLE
gcc complains about potentially uninitialized variable use

### ISSUE THREAD

**Issue by GoogleCodeExporter** [byzhang/rapidjson#17]
```
What steps will reproduce the problem?
1. Make a document.
2. use ParseInsitu to parse.
3. compile gcc -O3 -Wall

What is the expected output? What do you see instead?
no warnings.  I see reader.h:617: error: 'd' may be used uninitialized in this 
function
reader.h:556: error: 'i64' may be used uninitialized in this function.

What version of the product are you using? On what operating system?
0.1, Linux, gcc 4.6.2 -O3 -Wall

Please provide any additional information below.


```

Original issue reported on code.google.com by `brian.budge@gmail.com` on 22 Feb 2012 at 10:57

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r58.
```

Original comment by `milo...@gmail.com` on 28 Feb 2012 at 6:05
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r59.
```

Original comment by `milo...@gmail.com` on 28 Feb 2012 at 6:07

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 8668aebf662f.
```

Original comment by `milo...@gmail.com` on 6 Jun 2014 at 6:04
